webviz_subsurface/containers/_summary.py

```diff
@@ -0,0 +1,670 @@
+    get_time_series_statistics, get_time_series_fielgains
+
+
+# =============================================================================
+# Container
+# =============================================================================
+
+class Summary(WebvizContainer):
+    """Plot of time series data based on fmu-ensemble summary data.
+    Data are loaded from scratch an process via fmu-ensemble utilities.
+
+    Args:
+        ensembles: key-value-pait = ensemble-name: ensemble-path
+        column_keys: list = list of preselected vectors to be selectable
+        sampling: str = time-index / time-steps of summary-data
+    """
+
+    def __init__(
+            self,
+            app,
+            container_settings,
+            ensembles,
+            column_keys=None,
+            sampling: str = 'monthly',
+            base_ensembles: list = None,
+            delta_ensembles: list = None):
+
+        self.title = 'EnsembleSet'
+        self.uid = f'{uuid4()}'
+        self.time_index = sampling
+        self.column_keys = tuple(column_keys) if isinstance(
+            column_keys, (list, tuple)) else None
+        self.ensemble_paths = tuple(
+            (ensemble,
+             container_settings['scratch_ensembles'][ensemble])
+            for ensemble in ensembles)
+        self.base_ensembles = tuple(base_ensembles if base_ensembles else [
+            i[0] for i in self.ensemble_paths])
+        self.delta_ensembles = tuple(delta_ensembles if delta_ensembles else [
+            i[0] for i in self.ensemble_paths])
+        self.set_callbacks(app)
+
+    @property
+    def base_ens(self):
+        """ extracts base-ensemble from passed ensemble-combinations list"""
+        return self.ensemble_combinations[0]
+
+    @property
+    def delta_ens(self):
+        """ extracts delta-ensemble from passed ensemble-combinations list """
+        return self.ensemble_combinations[1]
+
+    @property
+    def dropwdown_vector_id(self):
+        """ component-id for vecotr selector"""
+        return f'dropdown-vector-{self.uid}'
+
+    @property
+    def chart_id(self):
+        """ component-id for chart => main plot"""
+        return f'chart-id-{self.uid}'
+
+    @property
+    def tab_id(self):
+        """ component-id for tab-selector"""
+        return f'_tab_id-{self.uid}'
+
+    @property
+    def chlst(self):
+        """ component-id for checklist """
+        return f'chlst-{self.uid}'
+
+    @property
+    def dropdown_iorens_id(self):
+        """ component-id for base-enseble-selector (fieldgaisn) """
+        return f'dropdown-iorens-{self.uid}'
+
+    @property
+    def dropdown_refens_id(self):
+        """ component-id for ensemble-selectors (fieldgains) """
+        return f'dropdown-refens-{self.uid}'
+
+    @property
+    def show_ens_selectors(self):
+        """ component-id for div including ensemble-selectors """
+        return f'show-ens-selectors-{self.uid}'
```

Comment:
This IDs above don't need to be `@property`-decorated functions. Instead they can be defined in `__init__` directly as e.g. `self.show_ens_selectors = f'show-ens-selectors-{self.uid}'` since all these properties are onlys used within the class.

Reply:
Changed,
did not know what is considered good style.